core/src/main/java/org/ehcache/spi/cache/tiering/CachingTier.java

```diff
@@ -47,6 +48,14 @@
   void invalidate(K key) throws CacheAccessException;
 
   /**
+   * Remove a mapping then call a function.
+   * @param key the key.
+   * @param invalidator the function to call.
+   * @throws CacheAccessException
+   */
+  void invalidate(K key, NullaryFunction<K> invalidator) throws CacheAccessException;
```

Comment:
I wouldn't name this param `invalidator`, it could do about anything... It's just a `NullaryFunction` invoked under lock scope (iirc, irrespectively of a mapping being there, right?)